hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java

```diff
@@ -286,13 +293,41 @@ private void processNonFeeTransfers(long consensusTimestamp, RecordItem recordIt
             if (aa.getAmount() != 0) {
                 NonFeeTransfer nonFeeTransfer = new NonFeeTransfer();
                 nonFeeTransfer.setAmount(aa.getAmount());
-                nonFeeTransfer.setId(new NonFeeTransfer.Id(consensusTimestamp, EntityId.of(aa.getAccountID())));
                 nonFeeTransfer.setPayerAccountId(recordItem.getPayerAccountId());
+
+                EntityId entityId = getEntityFromAccountId(aa.getAccountID());
+
+                nonFeeTransfer.setId(new NonFeeTransfer.Id(consensusTimestamp, entityId));
                 entityListener.onNonFeeTransfer(nonFeeTransfer);
             }
         }
     }
 
+    private EntityId getEntityFromAccountId(AccountID accountID) {
+        EntityId entityId;
+
+        switch (accountID.getAccountCase()) {
+            case ACCOUNTNUM:
+                entityId = EntityId.of(accountID);
+                break;
+            case ALIAS:
+                var optionalEntity = entityRepository.findByAlias(
+                        DomainUtils.toBytes(accountID.getAlias()));
+                if (!optionalEntity.isPresent()) {
+                    throw new InvalidDatasetException("AccountID not present for alias: " + accountID
+                            .getAlias());
+                }
+
+                entityId = EntityId.of(optionalEntity.get(), EntityType.ACCOUNT);
+                entityRepository.addToAliasCache(DomainUtils.toBytes(accountID.getAlias()), entityId.getId());
+                break;
+            default:
+                throw new InvalidDatasetException("Unsupported account: " + accountID.getAccountCase());
+        }
```

Comment:
Can short circuit return and use optional chaining to clean it up:

```suggestion
    private EntityId getEntityFromAccountId(AccountID accountID) {
        switch (accountID.getAccountCase()) {
            case ACCOUNTNUM:
                return EntityId.of(accountID);
            case ALIAS:
                var alias = DomainUtils.toBytes(accountID.getAlias());
                return entityRepository.findByAlias(alias)
                    .map(id -> EntityId.of(id, EntityType.ACCOUNT))
                    .orElseThrow(() -> new InvalidDatasetException("AccountID not found for alias: " + accountID));
            default:
                throw new InvalidDatasetException("Unsupported account: " + accountID);
        }
```

Reply:
Adopted